File: 1018-binary-prefix-divisible-by-5/1018-binary-prefix-divisible-by-5.java
class Solution {
    public List<Boolean> prefixesDivBy5(int[] nums) {
        List<Boolean> ans = new ArrayList<>();
        int mod = 0;   // store prefix modulo 5

        for (int bit : nums) {
            mod = (mod * 2 + bit) % 5;   // update only the remainder
            ans.add(mod == 0);
        }

        return ans;
    }
}

File: 1018-binary-prefix-divisible-by-5/1018-binary-prefix-divisible-by-5.py
class Solution:
    def prefixesDivBy5(self, nums: List[int]) -> List[bool]:
        num=0
        l=[]
        for i in nums:
            num=int((num*2+i)%5)
            if num%5==0:
                l.append(True)
            else:
                l.append(False)
        return l